<launch>
<!-- Gripper controller -->
  <node pkg="mechanism_control" type="spawner.py" args="$(find door_handle_detector)/gripper_effort_right.control" />

<!-- Cartesian trajectory controller -->
  <param name="cartesian_trajectory_right/autostart" type="bool" value="false" />
  <param name="cartesian_trajectory_right/root_name" type="string" value="torso_lift_link" />
  <param name="cartesian_trajectory_right/tip_name" type="string" value="r_gripper_tool_frame" />
  <param name="cartesian_trajectory_right/max_vel_trans" value="1.0" />
  <param name="cartesian_trajectory_right/max_vel_rot" value="1.0" />
  <param name="cartesian_trajectory_right/max_acc_trans" value="0.5" />
  <param name="cartesian_trajectory_right/max_acc_rot" value="0.8" />
  <param name="cartesian_trajectory_right/pose/p" value="20.0" />
  <param name="cartesian_trajectory_right/pose/i" value="0.1" />
  <param name="cartesian_trajectory_right/pose/d" value="0.0" />
  <param name="cartesian_trajectory_right/pose/i_clamp" value="0.5" />
  <param name="cartesian_trajectory_right/pose/twist/ff_trans" value="20.0" />
  <param name="cartesian_trajectory_right/pose/twist/ff_rot" value="5.0" />
  <param name="cartesian_trajectory_right/pose/twist/fb_trans/p" value="20.0" />
  <param name="cartesian_trajectory_right/pose/twist/fb_trans/i" value="0.5" />
  <param name="cartesian_trajectory_right/pose/twist/fb_trans/d" value="0.0" />
  <param name="cartesian_trajectory_right/pose/twist/fb_trans/i_clamp" value="1.0" />
  <param name="cartesian_trajectory_right/pose/twist/fb_rot/p" value="1.5" />
  <param name="cartesian_trajectory_right/pose/twist/fb_rot/i" value="0.1" />
  <param name="cartesian_trajectory_right/pose/twist/fb_rot/d" value="0.0" />
  <param name="cartesian_trajectory_right/pose/twist/fb_rot/i_clamp" value="0.2" />
  <node pkg="mechanism_control" type="spawner.py" args="$(find arm_life_test)/cartesian_trajectory_right.xml" output="screen"/>


<!-- Cartesian tff controller -->
  <param name="cartesian_tff_right/autostart" type="bool" value="false" />
  <param name="cartesian_tff_right/root_name" type="string" value="torso_lift_link" />
  <param name="cartesian_tff_right/tip_name" type="string" value="r_gripper_tool_frame" />
  <param name="cartesian_tff_right/twist_to_wrench_trans" value="10.0" />
  <param name="cartesian_tff_right/twist_to_wrench_rot" value="1.0" />
  <param name="cartesian_tff_right/vel_trans/p" value="2.0" />
  <param name="cartesian_tff_right/vel_trans/i" value="0.5" />
  <param name="cartesian_tff_right/vel_trans/d" value="0.0" />
  <param name="cartesian_tff_right/vel_trans/i_clamp" value="1.0" />
  <param name="cartesian_tff_right/vel_rot/p" value="0.3" />
  <param name="cartesian_tff_right/vel_rot/i" value="0.1" />
  <param name="cartesian_tff_right/vel_rot/d" value="0.0" />
  <param name="cartesian_tff_right/vel_rot/i_clamp" value="0.2" />
  <param name="cartesian_tff_right/pos_trans/p" value="10.0" />
  <param name="cartesian_tff_right/pos_trans/i" value="0.5" />
  <param name="cartesian_tff_right/pos_trans/d" value="0.0" />
  <param name="cartesian_tff_right/pos_trans/i_clamp" value="1.0" />
  <param name="cartesian_tff_right/pos_rot/p" value="0.5" />
  <param name="cartesian_tff_right/pos_rot/i" value="0.1" />
  <param name="cartesian_tff_right/pos_rot/d" value="0.0" />
  <param name="cartesian_tff_right/pos_rot/i_clamp" value="0.2" />
  <node pkg="mechanism_control" type="spawner.py" args="$(find arm_life_test)/cartesian_tff_right.xml" output="screen"/>

</launch>
